docs/development/design/refactor-remote-repository.rst

```diff
@@ -0,0 +1,101 @@
+======================================
+ Refactor ``RemoteRepository`` object
+======================================
+
+
+This document describes the current usage of ``RemoteRepository`` objects and proposes a new normalized modeling.
+
+
+Goals
+=====
+
+* De-duplicate data stored in our database.
+* Save only one ``RemoteRepository`` per GitHub repository.
+* Use an intermediate table between ``RemoteRepository`` and ``User`` to store associated remote data for the specific user.
+* Make this model usable from our SSO implementation.
+* Use Post ``JSONField`` to store associated ``json`` remote data.
+* Do not disconnect ``Project`` and ``RemoteRepository`` when a user delete/disconnects their account.
+
+
+Non-Goals
+=========
+
+* Keep ``RemoteRepository`` in sync with GitHub repositories.
+* Delete ``RemoteRepository`` objects deleted from GitHub.
+* Listen to GitHub events to detect ``full_name`` changes and update our objects.
+
+.. note::
+
+   We may need/want some of these non-goals in the future.
+   They are just outside the scope of this document.
+
+
+Current Implementation
+======================
+
+When a user connect their account to a social account, we create a
+
+* ``allauth.socialaccount.models.SocialAccount``
+  * basic information (provider, last login, etc)
+  * provider's specific data saved in a JSON under ``extra_data``
+* ``allauthsocialaccount.models.SocialToken``
+  * token to hit the API on behalf the user
+
+
+We *don't create* any ``RemoteRepository`` at this point.
```

Comment:
On corporate we are syncing in background at login (and/or social account connected) for SSO --may be useful to bring that pattern into community as well.